components/Op/OpDetailForm.js

```diff
@@ -414,7 +437,7 @@ class OpDetailForm extends Component {
                   {getFieldDecorator('location', {
                     rules: [
                       {
-                        required: true,
+                        required: this.state.requiredForPublish,
```

Comment:
As location is a select box with a default of online. It is not possible for this value to be missing.  Hence the validator is redundant.

Reply:
I can remove this, but the default is 'online', leaving the validation can prompt people to actively change, the more information the user inputs, the better?